IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m

```diff
@@ -218,6 +231,36 @@ - (nullable NSData *)decrypt:(nonnull NSString *)encryptedMessageString
     }
 }
 
+- (NSString *)signData:(NSString *)message
+{
+    NSData *signedData = [[message dataUsingEncoding:NSUTF8StringEncoding] msidSignHashWithPrivateKey:self.privateKeyRef];
+    return [[NSString alloc] initWithData:signedData encoding:NSUTF8StringEncoding];
+}
+
+- (NSDate *)creationDate
+{
+    if (!_creationDate)
+    {
+        NSDictionary* privateKeyQuery = @{ (id)kSecValueRef: (__bridge id)self.privateKeyRef,
+         (id)kSecClass: (id)kSecClassKey,
+         (id)kSecReturnAttributes:(id)kCFBooleanTrue
+        };
+        
+        CFDictionaryRef result = nil;
+        OSStatus status = SecItemCopyMatching((CFDictionaryRef)privateKeyQuery, (CFTypeRef *)&result);
+        
+        if (status != errSecSuccess)
+        {
+            return nil;
```

Comment:
If we fail to query the date, does it mean we'll run this query every time?

Reply:
As discussed, creationDate is always present as part of the keychain item's attributes.